src/main/diagnostics/diagnosticsManager.ts

```diff
@@ -1,36 +1,21 @@
 import * as vscode from 'vscode';
 import { ExtensionComponent } from '../extensionComponent';
 import { TreesManager } from '../treeDataProviders/treesManager';
-import { AbstractCodeActionProvider } from './abstractCodeActionProvider';
-import { GoCodeActionProvider } from './goCodeActionProvider';
-import { MavenCodeActionProvider } from './mavenCodeActionProvider';
-import { NpmCodeActionProvider } from './npmCodeActionProvider';
-import { PypiCodeActionProvider } from './pypiCodeActionProvider';
-import { YarnCodeActionProvider } from './yarnCodeActionProvider';
-import { ApplicabilityCodeActionProvider } from './applicabilityActionProvider';
+import { AbstractFileActionProvider } from './abstractFileActionProvider';
+import { DescriptorActionProvider } from './descriptorActionProvider';
 
 /**
  * In case of project descriptor (i.e package.json) open, perform:
  * 1. Populate the 'Problems' view with top severities of the project dependencies.
  * 2. Provide vulnerabilities icons on the left gutter white line under a dependency in the project descriptor.
- *
- * In case of source code file open, perform:
- * 1. Populate the 'Problems' view with top severities of the CVEs vulnerabilities.
- * 2. Provide red, yellow, green or white line under a vulnerable line in the source code file.
  */
 export class DiagnosticsManager implements ExtensionComponent {
-    private _codeActionProviders: (AbstractCodeActionProvider | ApplicabilityCodeActionProvider)[] = [];
+    private _codeActionProviders: AbstractFileActionProvider[] = [];
 
     constructor(treesManager: TreesManager) {
         let diagnosticCollection: vscode.DiagnosticCollection = vscode.languages.createDiagnosticCollection();
```

Comment:
the logic is the same for all those sub classes the only difference is when locating the position in the descriptor, this will reduce code duplication from all the sub classes